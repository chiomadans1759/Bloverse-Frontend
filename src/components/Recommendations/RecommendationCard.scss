@import 'sass-rem';

.recommendation-card {
  border-radius: 0 !important;
  height: rem(344px);
  width: rem(281px);
  padding: rem(4px);
  .card-body {
    padding: rem(12px);
    .card-title {
      color: #868686;
      font-size: rem(14px);
    }
  }

  &--list {
  width: 100%;
  border: none !important;
  margin: rem(0px);
  }

  &>img {
    border-radius: 0 !important;
  }
  &--content {
    height: rem(65px);
    &__text {
      height: 60%;
      overflow: hidden;
       a{
          color: #333333;
        }
    }
    &__username {
      margin: rem(8px) auto;
      .initials {
        height: rem(16px);
        width: rem(16px);
        font-size: rem(10px);
        text-align: center;
        padding: 0;
        padding-top: rem(1.5px);
        padding-bottom: rem(1.5px);
        background-color: #868686;
        border: none;
      }
      .initials {
        height: rem(16px);
        width: rem(16px);
        font-size: rem(10px);
        text-align: center;
        padding: 0;
        padding-top: rem(2.5px);
        padding-bottom: rem(1.5px);
        background-color: #868686;
        border: none;
      }
      .name {
        color: #565656;
        font-weight: 500;
        margin: auto rem(5px);
      }
      .name::after {
        content: '\2022';
        color: #2F80ED;
        margin-left: rem(5px);
        font-weight: 800;
      }
      .time {
        font-size: rem(12px);
        color: #868686;
      }
    }
  }
  .actions {
    height: rem(21px);
    margin: rem(12px) auto;
    margin-bottom: 0;
    .count {
      margin: rem(7px);
      color: #252525;
    }
    .views {
      color: #868686;
      padding: 0;
      cursor: pointer;
    }
    .others {
      margin:0;
      padding: 0;
      .likes {
        border-right: rem(1px) solid #e6e5e5;
        padding: 0;
        display: inline-flex;
        justify-content: space-around;
        color: #2F80ED;
        .thumbs-up, .thumbs-down {
          cursor: pointer;
        }
      }
      .comments {
        border-right: rem(1px) solid #e6e5e5;
        padding: 0;
        display: inline-flex;
        justify-content: space-around;
        align-items: center;
        color: #868686;
        .coment, .share {
          cursor: pointer;
        }
      }
      .ellipsis {
        cursor: pointer;
        font-size: rem(16px);
        padding: 0;
        text-align: center;
        color: #868686;
          .dropdown-toggle {
          #option-icon {
            display: inline-block;
            background-color: #bcbcbc;
            width: rem(3px);
            height: rem(3px);
            margin-left: rem(2px);
            border-radius: 50%;
          }
          &::after {
            display: none;
          }
        }
      }
    }
  }
}
.list-img{
  height: rem(109px);
  width: rem(144px);
}

.icon {
  color: #0083e6 !important;
}

@media screen and (max-width: 1190px) and (min-width: 978px) {
  .profile-card {
    width: rem(240px) !important;
  }
}

@media screen and (min-width: 1020px) {
  .profile-card {
    width: rem(240px) !important;
  }
  .recommendation-card {
    border-radius: 0 !important;
    height: rem(344px);
    width: rem(281px);
    padding: rem(4px);
    .card-body {
      padding: rem(12px);
      .card-title {
        color: #868686;
      }
    }
    &>img {
      border-radius: 0 !important;
    }
    &--content {
      height: rem(65px);
      &__text {
        height: 60%;
        overflow: hidden;
        font-size: rem(14px);
        color: #303030;
      }
      &__username {
        margin: rem(8px) auto;
        .initials {
          height: rem(16px);
          width: rem(16px);
          font-size: rem(10px);
          text-align: center;
          padding: 0;
          padding-top: rem(1.5px);
          padding-bottom: rem(1.5px);
          background-color: #868686;
          border: none;
        }
        .initials {
          height: rem(16px);
          width: rem(16px);
          font-size: rem(10px);
          text-align: center;
          padding: 0;
          padding-top: rem(2.5px);
          padding-bottom: rem(1.5px);
          background-color: #868686;
          border: none;
        }
        .name {
          color: #565656;
          font-weight: 500;
          margin: auto rem(5px);
        }
        .name::after {
          content: '\2022';
          color: #2F80ED;
          margin-left: rem(5px);
          font-weight: 800;
        }
        .time {
          font-size: rem(12px);
          color: #868686;
        }
      }
    }
    .actions {
      height: rem(21px);
      margin: rem(12px) auto;
      margin-bottom: 0;
      .count {
        margin: rem(7px);
        color: #252525;
      }
      .views {
        color: #868686;
        padding: 0;
        cursor: pointer;
      }
      .others {
        margin:0;
        padding: 0;
        .likes {
          border-right: rem(1px) solid #e6e5e5;
          padding: 0;
          display: inline-flex;
          justify-content: space-around;
          .thumbs-up, .thumbs-down {
            cursor: pointer;
          }
        }
        .comments {
          border-right: rem(1px) solid #e6e5e5;
          padding: 0;
          display: inline-flex;
          justify-content: space-around;
          align-items: center;
          color: #868686;
          .coment, .share {
            cursor: pointer;
          }
        }
        .ellipsis {
          cursor: pointer;
          font-size: rem(16px);
          padding: 0;
          text-align: center;
          color: #868686;
        }
      }
    }
  }
}
