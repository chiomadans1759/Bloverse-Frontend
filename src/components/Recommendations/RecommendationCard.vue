<template>
  <main :class="['card', 'recommendation-card', customClass]">
    <img src="@/assets/photo-4.jpg" class="card-img-top" alt="...">
    <div class="card-body">
      <h5 class="card-title">Category</h5>
      <div class="card-text recommendation-card--content">
        <p class="font-weight-bold recommendation-card--content__text">
          <router-link to="/posts/rere">Some quick example text to build on the card title and make up the bulk of the card's content.
          Some quick example text to build on the card title and make up the bulk of the card's content.
          Some quick example text to build on the card title and make up the bulk of the card's content.</router-link>
        </p>
        <p class="recommendation-card--content__username" v-if="type === 'consumer'">
          <span class="time">1 min ago</span>
        </p>
        <p class="recommendation-card--content__username" v-else>
          <a class="btn btn-primary initials" href="#" role="button"></a>
          <span class="font-weight-bold name">Johndoe</span>
          <span class="time">1 min ago</span>
        </p>
      </div>
      <div class="card-text row actions">
        <div class="col-7 views">
          <i class="fal fa-eye"></i>
          <span class="count">123</span>
        </div>
        <div class="col-5 row others centerEle">
          <div class="col-9 likes">
            <p class="thumbs-up centerEle" @click="toggleLike()" v-show="!likePost">
              <i class="fal fa-thumbs-up"></i>
              <span class="count">18</span>
            </p>
            <p class="thumbs-up centerEle" @click="toggleLike()" v-show="likePost">
              <i class="fas fa-thumbs-up"></i>
              <span class="count">18</span>
            </p>
            <p class="thumbs-down centerEle" @click="toggleDislike()" v-show="!dislikePost">
              <i class="fal fa-thumbs-down"></i>
              <span class="count">1</span>
            </p>
            <p class="thumbs-down centerEle" @click="toggleDislike()" v-show="dislikePost">
              <i class="fas fa-thumbs-down"></i>
              <span class="count">1</span>
            </p>
          </div>
          <div class="col-3 ellipsis">
            <span
              class="nav-link dropdown-toggle"
              href="#"
              id="navbarDropdown"
              role="button"
              data-toggle="dropdown"
              aria-haspopup="true"
              aria-expanded="false"
              @click="positionDropdownArrow"
            >
              <i class="fal fa-ellipsis-h"></i>
            </span>
            <div :class="['dropdown-menu', 'bl-dropdown-menu', { 'top-dropdown': !dropdownArrowDown , 'bottom-dropdown': dropdownArrowDown }]" aria-labelledby="dropdownMenuReference" ref="dropdown">
              <div class="d-flex bl-dropdown-menu__countries">
                <div class="favourite-countries">
                     <p class="dropdown-item country">Report Post</p>
                    <hr>
                    <hr>
                    <p class="dropdown-item country pt-3" v-show="isConsumer">
                      <i class="fal fa-times"></i>
                      <span class="count">Remove</span>
                    </p>
                    <p class="dropdown-item country">
                      <i class="fal fa-plus "></i>
                      <span class="count">Creators Profile</span>
                    </p>
                    <p class="dropdown-item country">
                      <i class="fal fa-comment-dots "></i>
                      <span class="count">Comment
                        <span class="badges">5</span>
                      </span>
                    </p>
                    <p class="dropdown-item country hidden-lg-down pb-3">
                      <i class="fal fa-share"></i>
                      <span class="count">Share
                        <span class="badges">5</span>
                      </span>
                    </p>
                    <p class="dropdown-item country toggle-bookmark"  v-show="!addBookmark && !isConsumer" @click="toggleBookmark()">
                      <i class="fal fa-bookmark"></i>
                      <span class="count">Read Later</span>
                    </p>
                    <p class="dropdown-item country toggle-bookmark" v-show="addBookmark && !isConsumer" @click="toggleBookmark()">
                      <i class="fas fa-bookmark"></i>
                      <span class="count">Read Later</span>
                    </p>
                    <hr>
                    <hr>
                    <p class="dropdown-item country toggle-story" v-show="!readStory" @click="toggleReadStory()">
                      <i class="fal fa-book-open"></i>
                      <span class="count">Read Story</span>
                    </p>
                    <p class="dropdown-item country toggle-story" v-show="readStory" @click="toggleReadStory()">
                      <i class="fas fa-book-open"></i>
                      <span class="count">Read Story</span>
                    </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
import { Carousel, Slide } from "vue-carousel";

export default {
  name: "RecommendationCard",
  props: {
    type: {
      type: String
    },
    customClass: {
      type: String,
      required: false
    },
    isConsumer: {
      type: Boolean,
      default: false
    },
  },
  data() {
    return {
      likePost: false,
      dislikePost: false,
      addBookmark: false,
      readStory: false,
      dropdownArrowDown: false
    };
  },
  components: { Carousel, Slide },
  methods: {
    toggleLike() {
      this.likePost = !this.likePost;
      this.dislikePost = false;
    },
    toggleDislike() {
      this.dislikePost = !this.dislikePost;
      this.likePost = false;
    },
    toggleBookmark() {
      this.addBookmark = !this.addBookmark;
    },
    toggleReadStory() {
      this.readStory = !this.readStory;
    },
    positionDropdownArrow() {
      const dropdownEle = this.$refs.dropdown;
      this.dropdownArrowDown = false;

      const dropdownPositon = dropdownEle.getAttribute('x-placement');
      if (dropdownPositon === 'top-start') {
        this.dropdownArrowDown = true;
      }
    }
  }
};
</script>

<style lang='scss' scoped>
@import "./RecommendationCard.scss";
</style>
