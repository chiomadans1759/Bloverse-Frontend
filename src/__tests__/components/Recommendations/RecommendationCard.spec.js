import { mount, RouterLinkStub  } from '@vue/test-utils'
import RecommendationCard from '@/components/Recommendations/RecommendationCard.vue';

describe('RecommendationCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(RecommendationCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should toggle like button', () => {
    const wrapper = mount(RecommendationCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    wrapper.findAll('.thumbs-up').at(0).trigger('click');

    expect(wrapper.vm.likePost).toBe(true);
    expect(wrapper.vm.dislikePost).toBe(false);
  });

  it('should toggle dislike button', () => {
    const wrapper = mount(RecommendationCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    wrapper.findAll('.thumbs-down').at(0).trigger('click');

    expect(wrapper.vm.likePost).toBe(false);
    expect(wrapper.vm.dislikePost).toBe(true);
  });

  it('should toggle the read later icon', () => {
    const wrapper = mount(RecommendationCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.toggle-bookmark').at(1).trigger('click');

    expect(wrapper.vm.addBookmark).toBe(true);
  });

  it('should toggle the read story icon', () => {
    const wrapper = mount(RecommendationCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.toggle-story').at(1).trigger('click');

    expect(wrapper.vm.readStory).toBe(true);
  });

  it('should change the arrow position of the dropdown', () => {
    const wrapper = mount(RecommendationCard, {
      stubs: { RouterLink: RouterLinkStub },
    });

    wrapper.findAll('.dropdown-toggle').at(0).trigger('click');

    expect(wrapper.vm.dropdownArrowDown).toBe(false);
  });
});
