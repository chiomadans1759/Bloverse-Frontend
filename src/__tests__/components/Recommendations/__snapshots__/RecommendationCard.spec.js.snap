// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationCard Component should render correctly 1`] = `
<main
  class="card recommendation-card"
>
  <img
    alt="..."
    class="card-img-top"
    src="@/assets/photo-4.jpg"
  />
   
  <div
    class="card-body"
  >
    <h5
      class="card-title"
    >
      Category
    </h5>
     
    <div
      class="card-text recommendation-card--content"
    >
      <p
        class="font-weight-bold recommendation-card--content__text"
      >
        <a>
          Some quick example text to build on the card title and make up the bulk of the card's content.
        Some quick example text to build on the card title and make up the bulk of the card's content.
        Some quick example text to build on the card title and make up the bulk of the card's content.
        </a>
      </p>
       
      <p
        class="recommendation-card--content__username"
      >
        <a
          class="btn btn-primary initials"
          href="#"
          role="button"
        />
         
        <span
          class="font-weight-bold name"
        >
          Johndoe
        </span>
         
        <span
          class="time"
        >
          1 min ago
        </span>
      </p>
    </div>
     
    <div
      class="card-text row actions"
    >
      <div
        class="col-7 views"
      >
        <i
          class="fal fa-eye"
        />
         
        <span
          class="count"
        >
          123
        </span>
      </div>
       
      <div
        class="col-5 row others"
      >
        <div
          class="col-9 likes"
        >
          <p
            class="thumbs-up"
          >
            <i
              class="fal fa-thumbs-up"
            />
             
            <span
              class="count"
            >
              18
            </span>
          </p>
           
          <p
            class="thumbs-up"
            style="display: none;"
          >
            <i
              class="fas fa-thumbs-up"
            />
             
            <span
              class="count"
            >
              18
            </span>
          </p>
           
          <p
            class="thumbs-down"
          >
            <i
              class="fal fa-thumbs-down"
            />
             
            <span
              class="count"
            >
              1
            </span>
          </p>
           
          <p
            class="thumbs-down"
            style="display: none;"
          >
            <i
              class="fas fa-thumbs-down"
            />
             
            <span
              class="count"
            >
              1
            </span>
          </p>
        </div>
         
        <div
          class="col-3 ellipsis"
        >
          <span
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            <i
              class="fal fa-ellipsis-h"
            />
          </span>
           
          <div
            aria-labelledby="dropdownMenuReference"
            class="dropdown-menu bl-dropdown-menu"
          >
            <div
              class="d-flex bl-dropdown-menu__countries"
            >
              <div
                class="favourite-countries"
              >
                <p
                  class="dropdown-item country"
                >
                  <i
                    class="fal fa-comment-dots"
                  />
                   
                  <span
                    class="count"
                  >
                    67
                  </span>
                </p>
                 
                <p
                  class="dropdown-item country"
                >
                  <i
                    class="fal fa-share"
                  />
                   
                  <span
                    class="count"
                  >
                    67
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
